rtb/analisi-requisiti/src/casi-uso.d/cliente-prenotazione.tex

```diff
@@ -1 +1,137 @@
-E
+\textbf{UC19-Prenotazione}
+\begin{itemize}
+    \item \textbf{Attore principale: }cliente.
+    \item \textbf{Precondizioni: }il cliente sta visualizzando le informazioni di un ristorante.
+    \item \textbf{Postcondizioni: }la prenotazione è salvata nella lista delle prenotazioni del cliente.
+    \item \textbf{Scenario principale:} 
+        \begin{enumerate}
+            \item Il cliente seleziona la funzionalità di richiesta di una nuova prenotazione;
+            \item Il sistema illustra i giorni in cui è possibile effettuare la richiesta;
+            \item Il cliente seleziona il giorno di calendario;
+            \item Il sistema illustra le fasce orarie disponibili;
+            \item Il cliente seleziona l'orario;
+            \item Il cliente inserisce il numero di persone che parteciperanno alla prenotazione;
+            \item Il cliente dà la conferma ed effettua la richiesta;
+            \item il cliente visualizza la lista delle sue prenotazioni,ove è presente la prenotazione appena
+            effettuata.
+        \end{enumerate}
+\end{itemize}
+
+%il cliente può visualizzare solo prenotazioni passate,quelle future...? le divido per categorie?
+%ci sta aggiungere un'estensione se non ci sono prenotazioni?tipo "il sistema comunica al cliente che la lista è vuota"
+\textbf{UC20-Visualizzazione lista prenotazioni}
+\begin{itemize}
+    \item \textbf{Attore principale: } cliente.
+    \item \textbf{Precondizioni: }il cliente sta visualizzando le sue informazioni personali.
+    \item \textbf{Postcondizioni: }il cliente visualizza la lista delle sue prenotazioni.
+    \item \textbf{Scenario principale:} 
+        \begin{enumerate}
+            \item L'utente seleziona la funzionalità di visualizzazione delle sue prenotazioni;
+            
+            \item L'utente visualizza la lista delle sue prenotazioni;
+        \end{enumerate}
+\end{itemize}
+
+\textbf{UC21-Visualizzazione singola prenotazione}
+\begin{itemize}
+    \item \textbf{Attore principale: } cliente.
+    \item \textbf{Precondizioni: }il cliente sta visualizzando la lista delle sue prenotazioni.
+    \item \textbf{Postcondizioni: }il cliente visualizza le informazioni relative ad una singola prenotazione.
+    \item \textbf{Scenario principale:} 
+        \begin{enumerate}
+            \item L'utente seleziona una prenotazione in particolare;
+            \item L'utente visualizza la lista delle informazioni relative alla prenotazione selezionata:
+            \begin{itemize}
+                \item Il nome del ristorante;
+                \item L'indirizzo del ristorante;
+                \item L'orario;
+                \item Il numero di persone ;
+                \item Lo stato della prenotazione (se è in attesa di conferma da parte del ristoratore
```

Comment:
```suggestion
                Lo stato della prenotazione che può essere nei seguenti stati:

-  In attesa
- Accettata
- Rifiutata
- In corso
- Pagata
- Annullata

```

Reply:
non è troppo uguale a quella di swenergy?

Reply:
Non so di cosa stai parlando XD
Comunque secondo me questo è il modo migliore per farla.
Non è molto sensato pensare di togliere la prenotazione rifiutata dalla lista perché altrimenti il cliente da un momento all'altro vede "scomparire" la prenotazione dalla lista.
Ok che viene notificato del cambio stato ma non mi sembra molto user-friendly

Reply:
eh lo so che non è user friendly ma come l'ho scritta è esattamente com'è stata pensata durante la discussione dei casi d'uso; ho rispettato la dicitura dell'estensione dei diagrammi UML: se c'è l'estensione di rifiuto del ristoratore vuol dire che il cliente non arriva alle post condizioni e cioè non riesce a vedere la prenotazione

Reply:
per il cliente la prenotazione può essere solo 'in attesa', 'accettata' o 'rifiutata'; 'annullata' è praticamente lo stesso stato di rifiutata ma la precondizione è che l'ha chiusa il cliente invece del ristoratore

Reply:
lo stato dei pagamenti lo vedi aprendo la prenotazione, ma da fuori la prenotazione dovrebbe risultare come 'accettata'

Reply:
Ok allora va bene